SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -168,30 +196,56 @@ bool operator<(const std::optional<Pokemon>& lhs, const std::optional<Pokemon>&
     if (!rhs.has_value()){
         return true;
     }
-    // For example, we could imagine that people want to sort first by ball type, then by dex number.
-    // if (lhs->ball_slug < rhs->ball_slug){
-    //     return true;
-    // }
-    // if (lhs->ball_slug > rhs->ball_slug){
-    //     return false;
-    // }
-    // if (lhs->shiny != rhs->shiny){
-    //     return lhs->shiny;
-    // }
-    // if (lhs->gmax != rhs->gmax){
-    //     return lhs->gmax;
-    // }
-    return lhs->dex_number < rhs->dex_number;
+
+    for (const BoxSortingSelection preference : *lhs->preferences){
+        switch(preference.sort_type) {
+            // NOTE edit when adding new struct members
+            // TODO TESTING and account for preference.reverse
+            case BoxSortingSortType::NationalDexNo:
+                return lhs->national_dex_number < rhs->national_dex_number;
+                break;
```

Comment:
if lhs->national_dex_number == rhs->national_dex_number, there is a logical bug because we'd like to break out of the switch instead of returning false directly (there may be more checks later on)

```suggestion
            case BoxSortingSortType::NationalDexNo:
                if (lhs->national_dex_number != rhs->national_dex_number){
                    return lhs->national_dex_number < rhs->national_dex_number;
                }
                break;
```